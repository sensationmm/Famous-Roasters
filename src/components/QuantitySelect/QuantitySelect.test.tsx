import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'

import { QuantitySelect } from '.'

global.alert = jest.fn()

describe('QuantitySelect component', () => {
  it('Renders correctly', async () => {
    const { container } = render(<QuantitySelect min={1} max={10} />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('The user can increment if is not yet on max', async () => {
    render(<QuantitySelect min={1} max={10} value={1} onChange={(v) => alert(v)} />)
    const button = await screen.findByTestId('quantity-plus')
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    const quantity = await screen.findByTestId('quantity-value')
    expect(quantity).toBeInTheDocument()
    expect(quantity.textContent).toBe('2')
  })

  it('The user cannot increment if max reached', async () => {
    render(<QuantitySelect min={1} max={10} value={10} onChange={(v) => alert(v)} />)
    const button = await screen.findByTestId('quantity-plus')
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    const quantity = await screen.findByTestId('quantity-value')
    expect(quantity).toBeInTheDocument()
    expect(quantity.textContent).toBe('10')
  })

  it('The user can decrement if is not yet on min', async () => {
    render(<QuantitySelect min={1} max={10} value={5} onChange={(v) => alert(v)} />)
    const button = await screen.findByTestId('quantity-minus')
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    const quantity = await screen.findByTestId('quantity-value')
    expect(quantity).toBeInTheDocument()
    expect(quantity.textContent).toBe('4')
  })

  it('The user cannot decrement if min reached', async () => {
    render(<QuantitySelect min={1} max={10} value={1} onChange={(v) => alert(v)} />)
    const button = await screen.findByTestId('quantity-minus')
    expect(button).toBeInTheDocument()
    fireEvent.click(button)
    const quantity = await screen.findByTestId('quantity-value')
    expect(quantity).toBeInTheDocument()
    expect(quantity.textContent).toBe('1')
  })
})
