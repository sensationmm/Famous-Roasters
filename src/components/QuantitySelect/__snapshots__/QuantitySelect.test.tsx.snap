// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantitySelect component Renders correctly 1`] = `
<div>
  <div
    class="inline-flex justify-between px-4 py-2 text-left bg-white rounded-full border border-coreUI-text-tertiary cursor-default"
    data-testid="quantity-select"
  >
    <button
      data-testid="quantity-minus"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="w-5 h-5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <div
      class="px-3.5"
      data-testid="quantity-value"
    >
      1
    </div>
    <button
      data-testid="quantity-plus"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="w-5 h-5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
